mne/io/edf/edf.py

```diff
@@ -474,35 +398,14 @@ def _get_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
         warn('Physical range is not defined in following channels:\n' +
              ', '.join(ch_names[i] for i in bad_idx))
         physical_ranges[bad_idx] = 1
-    if 'stim_data' in edf_info and stim_channel == 'auto':  # For GDF events.
-        cals = np.append(cals, 1)
-    if stim_channel is not None:
-        stim_channel = _check_stim_channel(stim_channel, ch_names, sel)
-
-    # Annotations
-    tal_ch_name = 'EDF Annotations'
-    tal_chs = np.where(np.array(ch_names) == tal_ch_name)[0]
-    if len(tal_chs) > 0:
-        logger.info('EDF annotations detected (consider using '
-                    'raw.find_edf_events() to extract them)')
-        if len(tal_chs) > 1:
-            warn('Channel names are not unique, found duplicates for: %s. '
-                 'Adding running numbers to duplicate channel names.'
-                 % tal_ch_name)
-        for idx, tal_ch in enumerate(tal_chs, 1):
-            ch_names[tal_ch] = ch_names[tal_ch] + '-%s' % idx
-    tal_sel = edf_info['sel'][tal_chs]
-    edf_info['tal_sel'] = tal_sel
-
-    if len(tal_sel) > 0 and stim_channel is not None and not preload:
-        raise RuntimeError('%s' % ('EDF+ Annotations (TAL) channel needs to be'
-                                   ' parsed completely on loading.'
-                                   ' You must set preload parameter to True.'))
+    stim_channel, stim_ch_name = \
+        _check_stim_channel(stim_channel, ch_names, sel)
```

Comment:
_check_stim_channel still accepts stuff that are not None or False for the stim_channel parameter but we have a RuntimeError above if it's not a bool or None. So _check_stim_channel should be simplified.

Reply:
@massich what do you think about this comment?

Reply:
Please mark resolved if this is resolved